### ISSUE TITLE
Legacy wrapper with rjson/RJSONIO behaviour

### ISSUE THREAD

**Issue by jeroenooms** [jeroenooms/jsonlite#73]
Many people that come from rjson or RJSONIO have complained about the different defaults. Maybe we should add a legacy wrapper to jsonlite that uses similar defaults as RJSONIO (but without the bugs).

 - auto_unbox = true
 - dataframe = "column"
 - na = "null"

That might make it easier for people to migrate. I think shiny does something similar internally.

**Comment by hadley** [comment]
I was thinking it might be use to have a few variants:

* rjson/RJSONIO legacy flavour
* API flavour, `auto_unbox = TRUE` and `dataframe = "column"`

**Comment by jeroenooms** [comment]
See also https://github.com/rstudio/shiny/issues/767.

**Comment by wch** [comment]
I think the command would now be something like:

```R
# RJSONIO compatibility wrapper
toJSON2 <- function(x, ...) {
  jsonlite::toJSON(I(x), dataframe = "columns", null = "null", na = "null",
                   auto_unbox = TRUE, use_signif = TRUE, force = TRUE,
                   rownames = FALSE, keep_vec_names = TRUE, ...)
}
```

**Comment by jeroenooms** [comment]
You need to include the arguments in `toJSON2` as well otherwise you specify them twice if the user does e.g. `toJSON2(x, dataframe="row")`

**Comment by wch** [comment]
Oh, right, thanks.

It turns out there's also an incompatible change with time conversion: rstudio/shiny#773.

**Comment by hadley** [comment]
Maybe `toApiJSON` would be a nice name?

**Comment by jeroenooms** [comment]
I'm not sure I want to promote the `RJSONIO` legacy behavior as optimal for apis. The `keep_vec_names = TRUE` option is dangerous and unnecessary and most API's that I have worked with prefer `dataframe="rows"`.

**Comment by hadley** [comment]
The auto boxing behaviour is __super__ annoying - I don't care about anything else

**Comment by jeroenooms** [comment]
Yes I am aware of that. But we should think the other defaults through as well if we want to get it right this time.

**Comment by wch** [comment]
OK, so the updated function for compatibility would look like:

```R
toJSON2 <- function(x, dataframe = "columns", null = "null", na = "null",
  auto_unbox = TRUE, use_signif = TRUE, force = TRUE, rownames = FALSE,
  POSIXt = "epoch", keep_vec_names = TRUE, ...) {

  jsonlite::toJSON(I(x), dataframe = dataframe, null = null, na = na,
    auto_unbox = auto_unbox, use_signif = use_signif, force = force,
    rownames = rownames, POSIXt = POSIXt, keep_vec_names = keep_vec_names, ...)
}
```

But for Shiny, it looks we're going to diverge from full RJSONIO compatibility a bit. Instead of `POSIXt="epoch"`, we're probably going to do `POSIXt="ISO8601", UTC = TRUE`.
